src/corehost/common/pal.h

```diff
@@ -50,16 +50,26 @@
 
 #endif
 
-
+// When running on a platform that is not supported in RID fallback graph (because it was unknown
+// at the time the SharedFX in question was built), we need to use a reasonable fallback RID to allow
+// consuming the native assets.
+//
+// For Windows and OSX, we will maintain the last highest RID-Platform we are known to support for them as the 
+// degree of compat across their respective releases is usually high.
+//
+// We cannot maintain the same (compat) invariant for linux and thus, we will fallback to using lowest RID-Plaform.
 #if defined(_WIN32)
 #define LIB_PREFIX
 #define MAKE_LIBNAME(NAME) (_X(NAME) _X(".dll"))
+#define FALLBACK_HOST_RID _X("win10")
```

Comment:
I'd opt for the most compatible thing.  That'd be the versioned RID since it still covers the un-versioned case.

> we would push others to follow a similar pattern and when that happens this wouldn't be a problem any longer

What if folks actually had a good reason for cross-compiling by OS version and didn't want to handle that at runtime?  The reason we are able to do this is we are putting forth a lot of effort to produce distro/version-agnostic binaries.  While its fine to recommend that to customers I don't think we should mandate it unless it can truly remove a lot of complexity.